### PR TITLE
[WAGON-485] ScpWagon file size Integer to Long

### PR DESCRIPTION
**Pull request by oassuncao**
Change filesize to Long as `resource.setContentLength`

### CODE REVIEW

**Comment by michael-o** [comment]
Is the value in bytes? If though, this truly can fail with intergers. Please open a JIRA issue for that.

**Comment by oassuncao** [comment]
If you check the `resource.setContentLength` this method receive a long value as a parameter. In the `ScpWagon` convert the integer to long (java cast). 

I only parse the value to Long, because if you have large file you will receive a error. My file has 3.8 GB in bytes is 3865470566. It is more than the capability of Integer Java

File: wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/jsch/ScpWagon.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public void fillInputData( InputData inputData )
                 throw new IOException( "Invalid transfer header: " + line );
             }
 
-            int filesize = Integer.valueOf( line.substring( 5, index ) ).intValue();
+            long filesize = Long.valueOf( line.substring( 5, index ) );
```
Comment (Tunaki) [inline wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/jsch/ScpWagon.java]: Small note, `Long.parseLong` is better here as it wouldn't involve an unboxing conversion.

Comment (michael-o) [inline wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/jsch/ScpWagon.java]: Good point. Waiting for this to be integrated and I will ran all tests against the patch.

**Comment by oassuncao** [comment]
Changed the code to use `Long.parseLong`

**Comment by michael-o** [comment]
Please squash them.

**Comment by oassuncao** [comment]
I squash the commits.

Thanks

**Comment by stephenc** [comment]
LGTM

**Comment by jira-importer** [comment]
Resolve #541

**Pull request closed without merging**